lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -82,18 +85,27 @@ def mock_get_score(self, score=(1.0, 2.0)):
         with patch("lms.djangoapps.grades.tasks.get_score", return_value=score):
             yield
 
-    def test_problem_weighted_score_changed_queues_task(self):
+    @patch('track.request_id_utils.uuid4')
+    def test_problem_weighted_score_changed_queues_task(self, uuid_mock):
         """
         Ensures that the PROBLEM_WEIGHTED_SCORE_CHANGED signal enqueues the correct task.
         """
         self.set_up_course()
         send_args = self.problem_weighted_score_changed_kwargs
+
+        # When a problem weighted score changes, the user action id/type are overridden.
+        # Here we override the uuid function to always return the same value generated
+        # in set_up_course(), and update the expected user action type to match
+        # a problem submission.
```

Comment:
Wouldn't the user action type be set to whatever you put in the request cache?  Why not set it in the request cache in `setUp`?